action.yml

```diff
@@ -1,56 +1,151 @@
+        if [ $((MINOR % 2)) -eq 0 ]; then
+          URL="https://pkgs.tailscale.com/stable/tailscale_${VERSION}_amd64.tgz"
+        else
+          URL="https://pkgs.tailscale.com/unstable/tailscale_${VERSION}_amd64.tgz"
+        fi
+        curl "$URL" -o tailscale.tgz
+        tar -C /tmp -xzf tailscale.tgz
+        rm tailscale.tgz
+        TSPATH=/tmp/tailscale_${VERSION}_amd64
+        sudo mv "${TSPATH}/tailscale" "${TSPATH}/tailscaled" /usr/bin
+    - name: Run Tailscale
+      shell: bash
+      env:
+        TAILSCALE_AUTHKEY: ${{ inputs.authkey }}
+        TAILSCALE_HOSTNAME: ${{ inputs.hostname }}
+        TAILSCALE_ACCEPT_DNS: ${{ inputs.acceptDNS }}
+        TAILSCALE_ACCEPT_ROUTES: ${{ inputs.acceptRoutes }}
+        TAILSCALE_ADVERTISE_EXIT_NODE: ${{ inputs.advertiseExitNode }}
+        TAILSCALE_ADVERTISE_ROUTES: ${{ inputs.advertiseRoutes }}
+        TAILSCALE_ADVERTISE_TAGS: ${{ inputs.advertiseTags }}
+        TAILSCALE_EXIT_NODE_ALLOW_LAN_ACCESS: ${{ inputs.exitNodeAllowLanAccess }}
+        TAILSCALE_EXIT_NODE: ${{ inputs.exitNode }}
+        TAILSCALE_FORCE_REAUTH: ${{ inputs.forceReauth }}
+        TAILSCALE_HOST_ROUTES: ${{ inputs.hostRoutes }}
+        TAILSCALE_NETFILTER_MODE: ${{ inputs.netfilterMode }}
+        TAILSCALE_QR: ${{ inputs.qr }}
+        TAILSCALE_SHIELDS_UP: ${{ inputs.shieldsUp }}
+        TAILSCALED_SOCKET: ${{ inputs.tailscaled_socket }}
+        TAILSCALED_STATE: ${{ inputs.tailscaled_state }}
+        TAILSCALED_TUN: ${{ inputs.tailscaled_tun }}
+        TAILSCALED_VERBOSE: ${{ inputs.tailscaled_verbose }}
+      run: |
+        if [ -z "${TAILSCALE_HOSTNAME}" ]; then
+          TAILSCALE_HOSTNAME="github-$(cat /etc/hostname)"
+        fi
+
+        up() {
+          sleep 5
+
+          sudo tailscale up \
+            --accept-dns=${TAILSCALE_ACCEPT_DNS:-true} \
+            --accept-routes=${TAILSCALE_ACCEPT_ROUTES:-false} \
+            --advertise-exit-node=${TAILSCALE_ADVERTISE_EXIT_NODE:-false} \
+            --advertise-routes=${TAILSCALE_ADVERTISE_ROUTES} \
+            --advertise-tags=${TAILSCALE_ADVERTISE_TAGS} \
+            --authkey=${TAILSCALE_AUTHKEY} \
+            --exit-node-allow-lan-access=${TAILSCALE_EXIT_NODE_ALLOW_LAN_ACCESS:-false} \
+            --exit-node=${TAILSCALE_EXIT_NODE} \
+            --force-reauth=${TAILSCALE_FORCE_REAUTH:-false} \
+            --host-routes=${TAILSCALE_HOST_ROUTES:-true} \
+            --hostname=${TAILSCALE_HOSTNAME:-$(cat /etc/hostname)} \
+            --login-server=${TAILSCALE_LOGIN_SERVER:-"https://login.tailscale.com"} \
+            --netfilter-mode=${TAILSCALE_NETFILTER_MODE:-on} \
+            --qr=${TAILSCALE_QR:-false} \
+            --shields-up=${TAILSCALE_SHIELDS_UP:-false} \
+            --snat-subnet-routes=${TAILSCALE_SNAT_SUBNET_ROUTES:-true}
+        }
+
+        if [ ! -d /dev/net ]; then mkdir /dev/net; fi
+        if [ ! -e /dev/net/tun ]; then mknod /dev/net/tun c 10 200; fi
+
+        up & sudo tailscaled \
```

Comment:
Should `tailscaled` be fired up before `tailscale up` here? This appears to be in the opposite order.